### PR TITLE
creeate dbmanager microservice

### LINKED ISSUE #93: Database migrations are manual and difficult to version
We should include a versioned migration system for our primary datastore `postgres` service. However, since the service can be used by multiple other services, the migration handler should reside outside of any language-specific ORM, and use plain SQL preferably.

### PR DESCRIPTION
**Pull request by AlexAppleManzer**
## Description
(#94 was the first attempt)

Another attempt at #93 that adds a new microservice that just runs migration scripts that it detects that hasn't been run yet.

This microservice is called dbhelper and hopefully will allow us in the future to do a lot more with this bot (maybe add cleanup cronjobs?)

## Checklist

- [x] [Linked PR to corresponding issue in this repo](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue)
- [x] Manually tested locally

fix #93

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -74,6 +74,17 @@ services:
         constraints: [node.role == manager]
     networks:
       - webnet
+  dbmanager:
+    image: johnyburd/dbmanager:${TAG:-latest}
+    build: ./dbmanager
+    environment:
+      PGUSER: autbot
+      PGPASSWORD: autism
+      PGDATABASE: autbot
```
Comment (johnyburd) [inline docker-compose.yml:83]: unless you need these exact names, use the variables in `architus.env` instead

File: dbmanager/dbmanager.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    with open(current_path, "w") as file:
+        print(f"Saving last-run migration '{last_migration}' to disk at {current_path}")
+        file.write(last_migration)
+
+
+def execute_sql_script(file):
+    """
+    Runes a sql script on the postgres container
+    """
+    process = subprocess.run(["psql", "-h", "postgres", "-U", "autbot", "-a", "-v", "ON_ERROR_STOP=1", "-f", file],
```
Comment (johnyburd) [inline dbmanager/dbmanager.py:57]: use env

File: docker-compose.yml (hunk 1 above)
Comment (AlexAppleManzer) [reply docker-compose.yml:83]: The only one I need is PGPASSWORD, but I might pull it out of here to keep it dry if i use enviromental variables anyways

**Comment by AlexAppleManzer** [comment]
> 
> 
> how is this intended to be used? should it be spun up independently of other services? or does it assume that the migrations can be pulled and run before the new version is built?

This should spin up with your postgres container and act as a companion container. The next PR after this will be to add cleanup scripts to this container that will run every time/in a cronjob

**Comment by johnyburd** [comment]
I'm getting an error deploying this with an existing db container
```postgres_1    | Stopping temporary Postgres server
postgres_1    | pg_ctl: PID file "/var/lib/postgresql/data/postmaster.pid" does not exist
postgres_1    | Is server running?
postgres_1    | Saving last-run migration '0000_seed.sql' to disk at /db/./current_migration/current.out
postgres_1    | 2020-06-20 04:12:54.008 UTC [7] FATAL:  database files are incompatible with server
postgres_1    | 2020-06-20 04:12:54.008 UTC [7] DETAIL:  The data directory was initialized by PostgreSQL version 12, which is not compatible with this version 13beta1 (Debian 13~beta1-1.pgdg100+1).```

**Comment by johnyburd** [comment]
put a placeholder file in `/dbmanager/migrations` or make the dockerfile create it

File: docker-compose.yml (hunk 1 above)
Comment (johnyburd) [reply docker-compose.yml:83]: please do

File: dbmanager/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
+FROM ubuntu:18.04
+
+RUN apt-get -y update
+RUN apt-get install -y postgresql-client
+RUN apt-get install -y python3
+RUN mkdir /app
+RUN mkdir /app/current_migration
+
+VOLUME /app/current_migration
```
Comment (johnyburd) [inline dbmanager/Dockerfile:9]: do you think it might make more sense to put this in `docker-compose.yml`?

Comment (AlexAppleManzer) [reply dbmanager/Dockerfile:9]: I'm not sure @jazevedo620 what do you think

Comment (AlexAppleManzer) [reply dbmanager/Dockerfile:9]: I think we should keep it here imo because no other service is using this volume, so we should keep it contained here.

**Pull request merged**